spec/support/matchers/have_annotations.rb

```diff
@@ -0,0 +1,43 @@
+RSpec::Matchers.define :have_annotations do |*expected|
+  actual_annotations = []
+
+  match do |actual|
+    actual_annotations = actual.annotations.map do |annotation|
+      {
+        prefix: annotation.key_prefix,
+        key: annotation.key_name,
+        value: annotation.value,
+      }
+    end
+
+    if expected.any?
+      expected = expected.map do |annotation|
+        {
+          prefix: annotation.with_indifferent_access[:prefix],
+          key: annotation.with_indifferent_access[:key],
+          value: annotation.with_indifferent_access[:value],
+        }
+      end
+
+      expect(actual_annotations).to match_array(expected)
+    else
```

Comment:
can these two paths be tested by match_array? if expected is empty then I would assume `expect(actual_annotations).to match_array([])` only passes if actual_annotations is empty. And also would assume that `map` on an empty array would yield an empty array as well. so the whole if/else could be simplified to just one path.

Reply:
I think this is a Ruby-ism because you can write:
```ruby
expect(thing).to have_annotations
```
Which you would expect to *pass* when there are any annotations. I wasn't able to differentiate between:
```ruby
expect(thing).to have_annotations
expect(thing).to have_annotations([])
```

The assumption is that if you want to assert on there being no annotations then you would write
```ruby
expect(thing).not_to have_annotations
```

That's the reason it appears unusual, and I wasn't able to find a better way. However I can at least comment.

Reply:
I've improved the messages and added a comment, so hopefully future maintainers will be clear on the original intention